R/shiny.R

```diff
@@ -502,12 +653,157 @@ ShinySession <- R6Class(
         ns = ns,
         makeScope = function(namespace) {
           self$makeScope(ns(namespace))
+        },
+
+        setBookmarkExclude = function(names) {
+          bookmarkExclude <<- names
+        },
+        getBookmarkExclude = function() {
+          bookmarkExclude
+        },
+        onBookmark = function(fun) {
```

Comment:
Do you think the order that `onBookmark(ed)`/`onRestore(d)` is invoked matters? Just curious.